accepted/automatic-redirects-by-default/automatic-redirects-by-default.md

```diff
@@ -0,0 +1,166 @@
+that is produced to the application's output directory.
+
+Following our conservative compatibility procedures we decided to make this
+feature opt-in rather than opt-out. Unfortunately, the opt-in wasn't based
+around the target framework the project uses at build-time, but based around the
+target-framework at creation time. That's because the setting that turns it on
+is conditionally emitted by the project template based on the selected target
+framework. If the project is upgraded later on, that setting is never turned on
+again. The net effect is that the vast majority of our customers have this
+setting still turned off.
+
+The proposal is to turn this setting on by default but allow application authors
+to turn the setting off if necessary. In other words, we change it from an opt-
+in feature to an opt-out feature.
+
+[binding redirects]: https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/redirect-assembly-versions
+[abrg]: https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/how-to-enable-and-disable-automatic-binding-redirection
+
+## Scenarios and User Experience
+
+### Consuming a .NET Standard 2.0 library from .NET Framework 4.6.1
+
+Michelle maintains a .NET Framework 4.5 application. She tries to install the
+new `System.IO.Pipelines` NuGet package that targets .NET Standard 2.0. She gets
+an error message from NuGet that this isn't compatible with .NET Framework 4.5.
+After some research, she learns that she must at least target .NET Framework
+4.6.1 in order to use .NET Standard 2.0 packages. After upgrading to .NET
+Framework 4.6.1, the package installs without errors. Michelle then starts to
+use pipelines from her app. When she F5s her application it just works fine.
+
+Michelle doesn't know that this is thanks to automatic binding redirect
+generation being on by default. Otherwise her application would have crashed
+with a `FileLoadException`.
+
+### Upgrading a NuGet dependency
+
+Robert is using FizzBuzz, a documentation framework that produces help files out
+of Markdown documents. FizzBuzz depends on Markdig, a popular .NET markdown
+library. Robert encounters an issue with some of his recent documentation
+changes which causes Markdig to crash. He notices that while there is no updated
+version of FizzBuzz, there is an updated version of Markdig that fixes his
+issue. Robert updates his (indirect) dependency on Markdig to the latest version
+that fixes the issue. He triggers a rebuild of the documentation and is happy to
+see that the crash no longer occurs.
+
+Robert doesn't know that this is thanks to automatic binding redirect generation
+being on by default. Otherwise his application would have crashed with a
+`FileLoadException`.
+
+## Requirements
+
+### Goals
+
+* **Turning this setting on must not break working applications**. We expect
+  this setting to make more applications work, but we must not break
+  applications where adding binding redirects breaks them.
```

Comment:
I think you need a little bit more on the risk assessment, e.g. when we talked in person, you explained why you didn't think it is likely to break many people

Reply:
I believe I have that covered in the [Q&A section](https://github.com/terrajobst/designs/blob/f1133dbb244f26b65d4fb4f085854c3af4f32ccf/accepted/automatic-redirects-by-default/automatic-redirects-by-default.md#why-is-the-setting-not-conditioned-on-the-target-framework)